repositories/strategies/routingStrategy.js

```diff
@@ -0,0 +1,287 @@
+  return trx("Routing_RuleSets")
+    .where(toDb({ id: routingRuleSetId, isDeleted: false }))
+    .first()
+    .then(fromDb);
+};
+
+const insertRoutingCondition = (trx, routingCondition) =>
+  trx("Routing_Conditions")
+    .insert(toDb(routingCondition))
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const insertRoutingRule = (trx, routingRule) =>
+  trx("Routing_Rules")
+    .insert(toDb(routingRule))
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const insertRoutingRuleSet = async (trx, questionPageId) =>
+  trx("Routing_RuleSets")
+    .insert(
+      toDb({
+        questionPageId: parseInt(questionPageId)
+      })
+    )
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+async function updateRoutingConditionStrategy(
+  trx,
+  { id: routingConditionId, questionPageId, answerId }
+) {
+  if (!isNil(answerId)) {
+    const answer = await findAnswerByIdAndQuestionPageId(
+      trx,
+      answerId,
+      questionPageId
+    );
+    if (!answer) {
+      throw new Error(
+        `Answer ${answerId} does not belong to ${questionPageId}. Choose a different answer.`
+      );
+    }
+  }
+
+  let firstAnswerOnPage;
+  if (isNil(answerId)) {
+    firstAnswerOnPage = await getFirstAnswer(trx, questionPageId);
+  }
+
+  const answerIdToStore = get(
+    firstAnswerOnPage,
+    "id",
+    isNil(answerId) ? null : answerId
+  );
+
+  await deleteRoutingConditionValues(trx, routingConditionId);
+  return updateRoutingCondition(
+    trx,
+    routingConditionId,
+    questionPageId,
+    answerIdToStore
+  );
+}
+
+const toggleConditionValueStrategy = async (
+  trx,
+  { conditionId, optionId, checked }
+) => {
+  const table = trx("Routing_ConditionValues");
+  const existing = await table.where(toDb({ conditionId, optionId }));
+  if (checked || isEmpty(existing)) {
+    return table
+      .insert(toDb({ conditionId, optionId }))
+      .returning("*")
+      .then(head)
+      .then(fromDb);
+  } else {
+    return table
+      .where(toDb({ conditionId, optionId }))
+      .del()
+      .returning("*")
+      .then(head)
+      .then(fromDb);
+  }
+};
+
+async function getAvailableRoutingDestinations(trx, pageId) {
+  const page = await getPageById(trx, pageId);
+  const section = await getSectionById(trx, page.sectionId);
+  const pageDestinations = await getPageDestinations(trx, page);
+  const sectionDestinations = await getSectionDestinations(trx, section);
+  return concat(pageDestinations, sectionDestinations);
+}
+
+function checkRoutingDestinations(
+  availableRoutingDestinations,
+  sectionDestination,
+  pageDestination
+) {
+  const partitionedDestinations = partition(
+    availableRoutingDestinations,
+    "sectionId"
+  );
+  const questionPages = first(partitionedDestinations);
+  const sections = last(partitionedDestinations);
+
+  if (isNil(pageDestination)) {
+    if (!find(sections, { id: parseInt(sectionDestination) })) {
```

Comment:
Actually, includes won't work here since it compares values and doesn't perform identity checking. https://stackoverflow.com/a/25171174. Have reverted back to `find` since it reads better than using `some`.

Reply:
Ah good catch. Didn't realise includes worked like that! Yeah, `find` is the best choice in that case